File: dynamic_programming/sum_of_no_adjacent_element/sum_of_no_adjacent_element.py
""" Solution Module """
class Solution:
    """ Solution Class """
    def run_solution(self, test_input):
        """ Solution Method """
        # [[1, 4, 5], [2, 0 0]]
        [matrix, n] = test_input

        # Sum including the current number
        inc = 0

        # Sum excluding the current number
        excl = 0

        for i in range(0, n):
            cur_max = max(inc, excl)
            inc = excl + max(matrix[0][i], matrix[1][i])
            excl = cur_max

        return max(excl, inc)

File: dynamic_programming/sum_of_no_adjacent_element/sum_of_no_adjacent_element.ts
export class Solution {
    /**
     * @param {string} testInput
     * @return {any} Returns either boolean to validate the test cases or a custom result.
     */
    run_solution(testInput: any) {
        const [matrix, n] = testInput;

        let included = 0;
        let excluded = 0;

        for (let i = 0; i < n; i++) {
            const current = Math.max(excluded, included);
            included = excluded + Math.max(matrix[0][i], matrix[1][i]);
            excluded = current;
        }

        return Math.max(excluded, included);
    }
}
